src/main/java/com/example/spring/services/UserServiceImpl.java

```diff
@@ -33,6 +36,10 @@ public UserResponseDto register(UserRegistrationRequestDto registrationRequestDt
         user.setPassword(passwordEncoder.encode(user.getPassword()));
         Role userRole = roleRepository.findByName(RoleName.ROLE_USER);
         user.setRoles(Set.of(userRole));
-        return userMapper.toDto(userRepository.save(user));
+        UserResponseDto savedUser = userMapper.toDto(userRepository.save(user));
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+        return savedUser;
```

Comment:
```suggestion
        shoppingCartService.createShoppingCart(user);
        return userMapper.toDto(user);
```

Where in ShopCartService

```
@Override
    public void createShoppingCart(User user) {
        ShoppingCart shoppingCart = new ShoppingCart();
        shoppingCart.setUser(user);
        shoppingCartRepository.save(shoppingCart);
    }
```